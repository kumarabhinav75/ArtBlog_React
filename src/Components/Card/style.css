.cardContainer {
  border: 1px solid darkgray;
  width: 380px;
  margin-top: 30px;
}
.art-image {
  max-width: 100%;
  max-height: 100%;
}

.art-image img {
  width: 100%;
  height: 300px
}

.date-time {
  display: flex;
  justify-content: space-between;
  margin: 10px 0px;
}

.date-time p{
  color: darkgray;
  margin: 0 10px;
  font-size: 0.8em;
}

.art-heading {
  font-size: 1.8em;
  color: black;
  margin: 10px 10px;
}

.art-story {
  color: black;
  margin: 10px 10px;
  font-size: 1.1em;
}

.art-reaction hr{
  width: 95%;
}
.clap-like {
  display: flex;
  justify-content: space-between;
}
.clap-like div {
  height:8%;
  width:8%;
  margin: 10px;
}

.claps  {
  display: flex;
  justify-content: flex-start;
}
.claps  img{
  width: 100%;
  height: 100%;
}

.claps span {
  color: #6d6c6c;
  margin: 0;
  vertical-align: middle;
  font-size: 1.3em;
  margin-left: 5px;
  margin-top: 0;
}