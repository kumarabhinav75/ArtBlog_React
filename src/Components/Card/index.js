import React, {Component} from 'react';
import './style.css';
import redHeart from '../../Icons/heart-red.svg';
import blackHeart from '../../Icons/heart-black.svg';

class Card extends Component{
  state = {
    liked: this.props.liked,
    clapCount: this.props.claps,
  };

  incrementClaps = () => {
    return this.setState({
      clapCount: this.state.clapCount+1,
    })
  }

  toggleLike = () => {
    return this.setState({
      liked: !(this.state.liked),
    })
  }

  flipLike = () => {
    if(this.state.liked === true) {
      return redHeart;
    }
    else {
      return blackHeart;
    }
  }
  render(){
    return(
      <div className="cardContainer">
        <section className="art-image">
          <img src={this.props.imageURL} alt="art"/>
        </section>
        <section className="date-time">
          <p>{this.props.date}</p>
          <p>{this.props.readingTime}</p>
        </section>
        <section>
          <p className="art-heading">{this.props.title}</p>
          <p className="art-story">{this.props.description}</p>
        </section>
        <section className="art-reaction">
          <hr/>
          <div className="clap-like">
            <div className="claps">
              <img src = {require('../../Icons/clapping.svg')} alt="clap" onClick={this.incrementClaps}/>
              <span>{this.state.clapCount}</span>
            </div>
            <div className="likes">
              <img src = {this.flipLike()} alt="clap" onClick = {this.toggleLike}/>
            </div>
          </div>
        </section>
      </div>
    );
  }
}

export default Card;