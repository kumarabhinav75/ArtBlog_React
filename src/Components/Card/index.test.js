import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Card from './index';

describe(Card, () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Card />);
    expect(tree).toMatchSnapshot();
  });

  it('should increment clapcount on click', () => {
    const wrapper = shallow(<Card claps={10} />);
    wrapper.instance().incrementClaps();
    expect(wrapper.instance().state.clapCount).toEqual(11);
    // console.log(wrapper.instance());
  });

  it('Should toggle inverse like', ()=> {
    const wrapper = shallow(<Card liked={true} />);
    wrapper.instance().toggleLike();
    expect(wrapper.instance().state.liked).toEqual(false);
  });
});