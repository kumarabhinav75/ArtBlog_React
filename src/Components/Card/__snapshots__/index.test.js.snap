// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<div
  className="cardContainer"
>
  <section
    className="art-image"
  >
    <img
      alt="art"
    />
  </section>
  <section
    className="date-time"
  >
    <p />
    <p />
  </section>
  <section>
    <p
      className="art-heading"
    />
    <p
      className="art-story"
    />
  </section>
  <section
    className="art-reaction"
  >
    <hr />
    <div
      className="clap-like"
    >
      <div
        className="claps"
      >
        <img
          alt="clap"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "clapping.svg",
            }
          }
        />
        <span />
      </div>
      <div
        className="likes"
      >
        <img
          alt="clap"
          onClick={[Function]}
          src="heart-black.svg"
        />
      </div>
    </div>
  </section>
</div>
`;
